radial_build_tools.py

```diff
@@ -0,0 +1,474 @@
+            visual_thickness = layer["thickness"]
+
+        comp_string = ""
+        if "composition" in layer:
+            comp_string = self.build_composition_string(layer["composition"])
+
+        description_str = ""
+        if "description" in layer:
+            description_str = textwrap.fill(
+                f'{layer["description"]}',
+                self.max_characters,
+                drop_whitespace=False,
+            )
+
+        # ensure sensible line breaks, this is the simplest way I have
+        # found due to how the above fields can be combined
+        text = f"{name}{thickness_str}\n{comp_string}{description_str}".rstrip()
+
+        newlines = text.count("\n")
+
+        min_thickness = (newlines + 1) * min_line_height
+
+        visual_thickness = min(
+            max(visual_thickness, min_thickness), self.max_thickness
+        )
+
+        return text, visual_thickness
+
+    def plot_radial_build(self):
+        """
+        Creates a radial build plot, with layers scaled between a minimum and
+        maximum pixel width to preserve readability.
+
+        Returns:
+            fig (matplotlib figure): figure containing radial build plot
+        """
+
+        char_to_height = 1.15
+        height = char_to_height * self.max_characters
+
+        # initialize list for lower left corner of each layer rectangle
+        ll = [0, 0]
+        fig = plt.figure(figsize=self.size)
+        plt.tight_layout()
+        ax = plt.gca()
+        ax.set_ylim(0, height + 1)
+
+        total_thickness = 0
+        for (name, layer), color in zip(self.build.items(), self.colors):
+
+            if layer.get("thickness") == 0:
+                continue
+
+            layer_str, visual_thickness = self.get_layer_string(name, layer)
+
+            ax.add_patch(
+                Rectangle(
+                    ll,
+                    visual_thickness,
+                    height,
+                    facecolor=color,
+                    edgecolor="black",
+                )
+            )
+
+            centerx = ll[0] + visual_thickness / 2 + 1
+            centery = height / 2
+            plt.text(
+                centerx,
+                centery,
+                layer_str,
+                rotation="vertical",
+                ha="center",
+                va="center",
+            )
+
+            ll[0] += float(visual_thickness)
+
+            total_thickness += visual_thickness
+
+        ax.set_xlim(-1, total_thickness + 1)
+        ax.set_axis_off()
+        plt.title(self.title)
+        return fig
+
+    def to_png(self, filename=None):
+        """
+        Write the plot to a png file.
+
+        Arguments:
+            filename (str): Optional, file name to write the plot to. If None,
+                file name will be the same as the plot title
+        """
+        if filename is None:
+            filename = self.title.replace(" ", "")
+
+        fig = self.plot_radial_build()
```

Comment:
This still mixes concerns a little.  The building of the plot should happen in a different method and then this one should just save it.  Perhaps `plot_radial_build()` should save the `fig` to a member variable `self.fig`?